src/structure/model.jl

```diff
@@ -222,378 +299,574 @@ end
+end
+
+
+"Get the occupancy of an `AbstractAtom`. Defaults to `1.0`."
+occupancy(at::Atom) = at.occupancy
+occupancy(dis_at::DisorderedAtom) = occupancy(defaultatom(dis_at))
+
+
+"Get the temperature factor of an `AbstractAtom`. Defaults to `0.0`."
+tempfac(at::Atom) = at.temp_fac
+tempfac(dis_at::DisorderedAtom) = tempfac(defaultatom(dis_at))
+
 
 """
-Get the element of an `AbstractAtom`. Defaults to `""` if not read from the PDB
-file.
+Get the element of an `AbstractAtom`. Defaults to `\"  \"`.
+`spaces` determines whether surrounding whitespace is retained (default
+`false`).
 """
-element(atom::Atom) = atom.element
+function element(at::Atom; spaces::Bool=false)
+    if spaces
+        return at.element
+    else
+        return strip(at.element)
+    end
+end
+
+element(dis_at::DisorderedAtom; spaces::Bool=false) = element(defaultatom(dis_at), spaces=spaces)
+
 
 """
-Get the charge on an `AbstractAtom`. Defaults to `""` if not read from the PDB
-file.
+Get the charge on an `AbstractAtom`. Defaults to `\"  \"`.
+`spaces` determines whether surrounding whitespace is retained (default
+`false`).
 """
-charge(atom::Atom) = atom.charge
+function charge(at::Atom; spaces::Bool=false)
+    if spaces
+        return at.charge
+    else
+        return strip(at.charge)
+    end
+end
+
+charge(dis_at::DisorderedAtom; spaces::Bool=false) = charge(defaultatom(dis_at), spaces=spaces)
+
+
+"Get the `Residue` that an `AbstractAtom` belongs to."
+residue(at::Atom) = at.residue
+residue(dis_at::DisorderedAtom) = residue(defaultatom(dis_at))
+residue(res::AbstractResidue) = res
 
 
 """
-Determines if an `AbstractAtom` represents a hetero atom, i.e. came from a
-HETATM record, or if an `AbstractResidue` represents a hetero molecule, i.e.
-consists of HETATM records.
+Determines if an `AbstractAtom` represents a non-hetero atom, e.g. came from an
+ATOM record in a Protein Data Bank (PDB) file.
 """
-ishetero(atom::AbstractAtom) = ishetatom(atom)
+ishetatom(at::Atom) = ishetres(residue(at))
+ishetatom(dis_at::DisorderedAtom) = ishetatom(defaultatom(dis_at))
+
 
 """
 Determines if an `AbstractAtom` is a `DisorderedAtom`, i.e. if there are
 multiple locations present for an atom.
 """
-isdisorderedatom(::Atom) = false
+isdisorderedatom(at::AbstractAtom) = isa(at, DisorderedAtom)
+
 
 """
-Get a descriptive atom ID for an `AbstractAtom` as a `Tuple` of the form
-(full residue ID, residue name, atom name).
+Get the alternative location ID of the default `Atom` in a `DisorderedAtom`.
+The default is the highest occupancy, or lowest character alternative location
+ID for ties.
 """
-atomid(atom::Atom) = (resid(atom; full=true), resname(atom), atomname(atom))
+defaultaltlocid(dis_at::DisorderedAtom) = dis_at.default
+
+
+# Constructor acts as a setter for the default alt loc ID
+function DisorderedAtom(dis_at::DisorderedAtom, default::Char)
+    @assert default in altlocids(dis_at) "The new default alternative location ID must be present in the atom"
```

Comment:
`@assert` should not be used for checking function arguments.  `throw(ArgumentError(message))` would be better in general.